site/content/docs/testapps/juiceshop.md

```diff
@@ -0,0 +1,118 @@
+---
+title: "OWASP Juice Shop"
+type: page
+zap-auth: 3
+maint: 3
+---
+
+### Overview
+
+OWASP Juice Shop is "probably the most modern and sophisticated insecure web application".
+
+It is open source, well maintained by independent volunteers, and [well documented](https://help.owasp-juice.shop/).
+
+* Home: https://owasp.org/www-project-juice-shop/
+* Online: https://preview.owasp-juice.shop/#/
+* Repo: https://github.com/juice-shop/juice-shop
+* Docker: https://hub.docker.com/r/bkimminich/juice-shop
+
+### Potential Pitfalls
+
+This is an online app which may be unavailable or broken at any point.
+Running a local version will give more consistent results.
+
+The UI has a set of pop-ups that are displayed when accessed via the new browser profile:
+
+* Welcome panel
+* Cookie approval dialog
+* Language notification (automatically disappears)
+
+These can all interfere with replaying authentication recordings.
+
+### Authentication
+
+Known users (username / password):
+
+* `admin@juice-sh.op` / `admin123`
+
+> [!IMPORTANT]
+> You should register a unique user with the online Juice Shop as other people may well change the default user.
+
+#### Browser Based Authentication
+
+ZAP can successfully authenticate to this app using 
+[Browser Based Authentication](/docs/desktop/addons/authentication-helper/browser-auth/) 
+and will successfully identify the session handling and verification.
+
+#### Client Script Authentication
+
+ZAP can successfully authenticate to this app using 
+[Client Script Authentication](/docs/desktop/addons/authentication-helper/client-script/) 
+and will successfully identify the session handling and verification.
+
+However you do need to record with either a new browser profile or using private/incognito mode so that you need to click
+on the pop-ups mentioned above. You also need to wait for the Language notification to disappear.
+
+### Recommended Environment
+
+```yaml
+env:
+  contexts:
+  - name: "Juice Shop"
+    urls:
+    - "https://preview.owasp-juice.shop"
+    includePaths:
+    - "https://preview.owasp-juice.shop.*"
+    excludePaths: [] 
+    authentication:
+      method: "browser"
+      parameters:
+        loginPageUrl: "https://preview.owasp-juice.shop/#/login"
+        browserId: "firefox-headless"
+        loginPageWait: 5
+      verification:
+        method: "poll"
+        loggedInRegex: "\\Q 200 OK\\E"
+        loggedOutRegex: "\\Q 304 Not Modified\\E"
```

Comment:
Really? Doesn't that indicate a local cache hit?

Reply:
What do you mean by local?